src/SplitClient.tsx

```diff
@@ -144,7 +146,7 @@ function SplitClient(props: ISplitClientProps) {
         // getSplitSharedClient is idempotent like factory.client: it returns the same client given the same factory, Split Key and TT
         const client = factory ? getSplitSharedClient(factory, props.splitKey, props.trafficType) : null;
         return (
-          <SplitComponent {...props} factory={factory} client={client} />
+          <SplitComponent {...props} factory={factory} client={client} attributes={props.attributes} />
```

Comment:
Regarding the `attributes` prop, we have to check if we should apply some memoization over it (a technique for optimizing component re-renders). Same for the `attributes` param in hooks. But probably not.